Added pros and cons of having staking proxy, controller or both

filippoweb3 opened the pull request:
- added pros cons of having staking proxy, controller or both
- prettified

Commit:
Added content

- added pros cons of having staking proxy, controller or both
- prettified

alfarok commented on docs/learn/learn-staking-advanced.md:
```diff
@@ -51,31 +51,46 @@ to staking, session and utility pallets.
+performed less often, i.e. bonding more funds and changing the controller (see figure above). The
+controller is used to sign for those staking actions that are performed more often such as
+nominating. Remember, staking on polkadot is not a set-and-forget action, as a nominator you will
+need to monitor the performance of your validators and change them if needed. Also, each time you
```
```suggestion
nominating. Remember, staking on polkadot is not a set-and-forget action, as a nominator you will
need to monitor the performance of your validators and make changes if needed. Also, each time you
```

alfarok commented on docs/learn/learn-staking-advanced.md:
```diff
@@ -51,31 +51,46 @@ to staking, session and utility pallets.
-Note that this situation is similar to having the controller as a staking proxy.
+If the stash and controller are different accounts, the staking proxy will just be used to bond more
+funds and change the controller. Thus the staking proxy will be used to sign for those transactions
+used less often and that are usually signed by the stash.
```
```suggestion
that are used less often and that are usually signed by the stash.
```

alfarok commented on docs/learn/learn-staking-advanced.md:
```diff
@@ -51,31 +51,46 @@ to staking, session and utility pallets.
+
+From a practical perspective we now have two accounts and we need to remember two passwords. From a
+security perspective who wants to fully control our staking actions must have access to two
+accounts.
```
```suggestion
From a practical perspective, we now have two accounts and we need to remember two passwords. From a
security perspective, the party who wants to fully control our staking actions must have access to two
accounts.
```

alfarok commented on docs/learn/learn-staking-advanced.md:
```diff
@@ -381,20 +396,23 @@ already maxed out.
-token liquid. {{ polkadot: DOT :polkadot }}{{ kusama: KSM :kusama }} is inflated according to the 
-system staking rate of the entire network.
+and the remainder going to the treasury. The current token supply on Kusama is
+{{ kusama: <RPC network="kusama" path="query.balances.totalIssuance" defaultValue={12619256191792480093}/> :kusama }}
```
```suggestion
and the remainder going to the treasury. The current token supply on Kusama is <RPC network="kusama" path="query.balances.totalIssuance" defaultValue={12619256191792480093}/>
```

This entire paragraph is already wrapped in conditonal rendering so you don't need to wrap the RPC component by itself here as it will actually mess up the formatting.

Before:
![image](https://user-images.githubusercontent.com/13341935/196609550-eea87d5e-f6b1-4910-896f-1a61bc210c50.png)

After:
![image](https://user-images.githubusercontent.com/13341935/196609651-5f7c45af-0cda-4eaf-a112-ab7b9d1d61ec.png)

alfarok commented on docs/learn/learn-staking-advanced.md:
```diff
@@ -408,29 +426,34 @@ estimation of all DOT that should be staked, per parachain slot).
-49 parachain slots. When the number of slots goes to 60, the ideal staking rate will be 45%. 
-[Here](https://github.com/paritytech/polkadot/blob/master/runtime/kusama/src/lib.rs#L535)
-is the code for reference. This code assumes that the number of slots auctioned correspond to the number of parachains on the relaychain, which may not be true as new slots can be occupied by old parachains that are renewing their lease. You can also track the progress on resolving this specific issue 
+At the time of updating this section, the ideal staking rate on Kusama was 50.5% when there were 49
```
Maybe use a date here instead for clarity.

filippoweb3 replied:
Now that I think about it, probably better to remove this dependence on the date of writing the section. I rewrote  it in more general terms.

alfarok reviewed (commented):
Hey @filippoweb3, looks good to me!  Lots of added clarity in various sections.  I just have a few optional suggestions that are probably more subjective so I will let you decide which ones (if any) you want to include.  The only required change is around the RPC component that current has nested conditional rendering and will produce formatting issues.  Let me know if you have any question!

filippoweb3 committed:
Update docs/learn/learn-staking-advanced.md

Co-authored-by: Keith Alfaro <keith4895@gmail.com>

filippoweb3 committed:
Update docs/learn/learn-staking-advanced.md

Co-authored-by: Keith Alfaro <keith4895@gmail.com>

filippoweb3 committed:
Update docs/learn/learn-staking-advanced.md

Co-authored-by: Keith Alfaro <keith4895@gmail.com>

filippoweb3 committed:
Update docs/learn/learn-staking-advanced.md

Co-authored-by: Keith Alfaro <keith4895@gmail.com>

Commit:
Minor edits

- removed staking percentage dependent on the date of writing the paragraph
- prettified

filippoweb3 commented:
@alfarok I need the approval to merge this PR 👍
